lib/ddtrace/contrib/sinatra/framework.rb

```diff
@@ -0,0 +1,78 @@
+module Datadog
+  module Contrib
+    # Instrument Sinatra.
+    module Sinatra
+      # Sinatra framework code, used to essentially:
+      # - handle configuration entries which are specific to Datadog tracing
+      # - instrument parts of the framework when needed
+      module Framework
+        # Configure Rack from Sinatra, but only if Rack has not been configured manually beforehand
+        def self.setup
+          Datadog.configure do |datadog_config|
+            sinatra_config = config_with_defaults(datadog_config)
+            activate_rack!(datadog_config, sinatra_config) unless Datadog.configuration.instrumented_integrations.key?(:rack)
+          end
+        end
+
+        def self.config_with_defaults(datadog_config)
+          datadog_config[:sinatra]
```

Comment:
I picked that method from Rails and simplified it but I am not sure if it's a good way to do it.